### DISCUSSION TITLE
Problem restarting nginx during installation after adding searxng.conf to sites-available.

### DISCUSSION THREAD

**Discussion by rjc3rd** [searxng/searxng#3540] [Q&A] [answered]
The app works and I was able to test in the browser when I installed on my local machine. Now that I decided to install on a virtual machine, I can test with curl and it is also working via ssh.

What I'm trying to do is...
I installed Debian 12 on qemu/kvm
In my hosts file I added the line...
192.168.155.21 searxng.vm
Via ssh I followed the directions here...
https://docs.searxng.org/admin/installation-searxng.html
After installing nginx before I add searxng.conf to sites-available, I test that I can reach the default page from my local machine at http://searxng.vm and it loads the default nginx page, showing that the nginx server is indeed serving pages.

After copying the sample content into the searxng.conf file in sites-available, and creating the symlink to sites-enabled...

nginx fails to restart.

I remove the new file searxng.conf and the symlink and restart just nginx just fine.

I recreate searxng.conf and test nginx config with...
sudo nginx -t
2024/06/05 20:28:44 [emerg] 962#962: "location" directive is not allowed here in /etc/nginx/sites-enabled/searxng.conf:1
nginx: configuration file /etc/nginx/nginx.conf test failed

I read on the nginx documentation that location directive needs to be inside of server directive.

When I add the server directive the test passes and states that the syntax is ok, test is successful

I restart nginx and uwsgi with no errors.

http://searxng.vm says 502 Bad Gateway
http://192.168.155.21 loads the default nginx page.

I don't know what info I need to provide in the server directive to load the site at http://searxng.vm.

I also think it may be related to the fact that I am not using SSL or https. I can't since it's on a VM on a local machine, unless I create a self-signed SSL, which I haven't looked in to yet because I'm not even sure if that's the problem yet.

Has anyone else done it like this and could I get some advise please.

**Comment by rjc3rd** [comment]
Here is my current /etc/nginx/sites-available/searxng.conf file.

    server {
    listen       80;
    server_name  searxng.vm;
    root /usr/local/searxng/searxng-src;

    location /searxng {

    uwsgi_pass unix:///usr/local/searxng/run/socket;

    include uwsgi_params;

    uwsgi_param    HTTP_HOST             $host;
    uwsgi_param    HTTP_CONNECTION       $http_connection;

    # see flaskfix.py
    uwsgi_param    HTTP_X_SCHEME         $scheme;
    uwsgi_param    HTTP_X_SCRIPT_NAME    /searxng;

    # see limiter.py
    uwsgi_param    HTTP_X_REAL_IP        $remote_addr;
    uwsgi_param    HTTP_X_FORWARDED_FOR  $proxy_add_x_forwarded_for;
    }
    }

With this, I am able to restart the ngnix server.

When I go to http://192.168.155.21
it loads default nginx html file correctly.

When I go to http://searxng.vm
I get error message 403 Forbidden

When I go to http://searxng.vm/searxng
I get error message 502 Bad Gateway

I believe that the correct address that would work with my current settins is, https://searxng.vm/searxng

Am I right to think that searxng requires a secure connection and rejects http requests?

I only want this to work on my local machine so I don't need SSL and it will never be exposed to internet.

Thanks.

**Comment by rjc3rd** [comment]
I followed the instructions on this site to install self-signed SSL.
https://www.howtogeek.com/devops/how-to-create-and-use-self-signed-ssl-on-nginx/

The SSL does work but with a warning that it is not secure due to the fact that it is self-signed but the browser does redirect to https now.

Still getting 403 Forbidden and 502 Bad Gateway though.

OK, now I wait for help. LOL

Thanks.

**Comment by rjc3rd** [comment]
My current searxng.conf file after installing the SSL.

    server {
    listen       443 ssl;
    include snippets/self-signed.conf;
    server_name  searxng.vm;
    root /usr/local/searxng/searxng-src;
    
    location /searxng {
    
    uwsgi_pass unix:///usr/local/searxng/run/socket;
    
    include uwsgi_params;
    
    uwsgi_param    HTTP_HOST             $host;
    uwsgi_param    HTTP_CONNECTION       $http_connection;
    
    # see flaskfix.py
    uwsgi_param    HTTP_X_SCHEME         $scheme;
    uwsgi_param    HTTP_X_SCRIPT_NAME    /searxng;
    
    # see limiter.py
    uwsgi_param    HTTP_X_REAL_IP        $remote_addr;
    uwsgi_param    HTTP_X_FORWARDED_FOR  $proxy_add_x_forwarded_for;
    }
    }
    
    server {
    listen 80;
    server_name searxng.vm;
    return 302 https://$server_name$request_uri;
    }

In case there is something wrong in here.

Thanks

**Comment by return42** [comment]
@rjc3rd we have a really active and help full community at [#searxng:matrix.org](https://matrix.to/#/#searxng:matrix.org) .. I recommend you to ask & discuss questions about the setup there. There is also more NGINX know than here in the SearXNG core development.

**Comment by rjc3rd** [comment, marked as answer]
I figured it out. I got it working this morning.
I answered the question over here where I found the mistake.
https://github.com/searxng/searxng/issues/3544
